src/aws-cpp-sdk-core/source/client/AWSClient.cpp

```diff
@@ -874,24 +877,53 @@ Aws::String Aws::Client::GetAuthorizationHeader(const Aws::Http::HttpRequest& ht
     return authHeader.substr(signaturePosition + strlen(Aws::Auth::SIGNATURE) + 1);
 }
 
-void AWSClient::BuildHttpRequest(const Aws::AmazonWebServiceRequest& request,
-    const std::shared_ptr<HttpRequest>& httpRequest) const
+void AWSClient::BuildHttpRequest(const Aws::AmazonWebServiceRequest& request, const std::shared_ptr<HttpRequest>& httpRequest) const
 {
-    //do headers first since the request likely will set content-length as it's own header.
+    //do headers first since the request likely will set content-length as its own header.
     AddHeadersToRequest(httpRequest, request.GetHeaders());
     AddHeadersToRequest(httpRequest, request.GetAdditionalCustomHeaders());
 
-    AddChecksumToRequest(httpRequest, request);
-
     if (request.IsEventStreamRequest())
     {
         httpRequest->AddContentBody(request.GetBody());
     }
     else
     {
-        AddContentBodyToRequest(httpRequest, request.GetBody(), request.ShouldComputeContentMd5(), request.IsStreaming() && request.IsChunked() && m_httpClient->SupportsChunkedTransferEncoding());
+        //Check if compression is required
+        if (Aws::Client::CompressionAlgorithm::NONE !=
+            request.GetSelectedCompressionAlgorithm(m_requestCompressionConfig)) {
```

Comment:
store a result of
request.GetSelectedCompressionAlgorithm in a variable to avoid calling it 3 times

Reply:
done